### PR TITLE
fix(prompt): prompt works with windows eol and eof (#8146)

### PR DESCRIPTION
**Pull request by sanderhahn**
Fixes issue #8146

The test case seems to check the Windows EOL "\r\n", however this is an illusion because somehow the test runner already translates the Windows into Unix EOL. Have tested the functionality interactively and had to remove the `isatty` return null to do so.

### CODE REVIEW

**Comment by sanderhahn** [comment]
Ah i see that `isatty` is also used in the other functions... removed it because you can't use prompt and pipe from stdin.
What do you think? @kt3k

**Review by lucacasonato — COMMENTED** [review]
Looks good. Could you add a test confirming the correct behaviour for both `alert` and `confirm` too?

File: cli/rt/41_prompt.js [hunk 1]
```diff
@@ -31,10 +32,6 @@
   function prompt(message = "Prompt", defaultValue) {
     defaultValue ??= null;
 
-    if (!isatty(stdin.rid)) {
-      return null;
-    }
-
```
Comment (lucacasonato) [inline cli/rt/41_prompt.js:37]: Can you revert this change? It looks like is unrelated to the goal of this PR.

**Comment by sanderhahn** [comment]
Done :+1:

**Comment by lucacasonato** [comment]
Thanks. Could you add a test confirming the correct behaviour for both `alert` and `confirm` too?

**Comment by sanderhahn** [comment]
Both these funtions call the same `readLineFromStdinSync` where the changes are made. As i said the test is only an illusion because it seems that "\r\n" is already converted to "\n" by the test runner. Couldn't pinpoint where this is happening but maybe its some rust code that already makes sure that EOL is cross platform. Have confirmed correct behavior by using an a tty and piped stdin because i don't have Windows available.

**Comment by lucacasonato** [comment]
Ah ok. We are going to need to figure out what is converting removing the carriage return here. I'll take a look.

**Comment by sanderhahn** [comment]
Maybe its possible to do binary io in the test runner...

**Comment by sanderhahn** [comment]
Think i found it: https://github.com/denoland/deno/blob/master/test_util/src/lib.rs#L917

**Comment by lucacasonato** [comment]
Oh we only test this test on unix. The test is prefixed with `#[cfg(unix)]`.

cc @bartlomieju want to land without working test?

**Comment by lucacasonato** [comment]
@sanderhahn This test gets never actually run on Windows. The `pty` crate we are using is unix only. We need to find a different way to test this on windows.

**Comment by ebebbington** [comment]
Subprocess to run a script (or one per prompt) and assert the output? probably not the best idea but just trying to offer any.
 help

**Review by lucacasonato — APPROVED** [review]
Discussed this offline. We are going to land this without a working Windows test, because setting up tty testing on Windows is out of scope for this PR.

Preemptively approving, but going to test on Windows that `https://deno.land/posts/v1.5/alert_confirm_prompt.js` example actually works. Thanks @sanderhahn

**Comment by lucacasonato** [comment]
@wperron confirmed this is indeed working now.
